MINOR: Map `mkString` format updated to default java format

kamalcph opened the pull request:
This is a minor change but it helps to improve the log readability.

kamalcph committed:
MINOR: Map `mkString` format updated to default java format

ijuma commented on clients/src/main/java/org/apache/kafka/common/utils/Utils.java:
```diff
@@ -351,7 +351,7 @@ public static String formatAddress(String host, Integer port) {
     }
 
     public static <K, V> String mkString(Map<K, V> map) {
```
Seems like we should just remove this method if it's the same as `Map.toString()`.

kamalcph replied:
Yes, I thought of the same. But, it's used in many places. Shall I remove it ?

ijuma replied:
Yes, please.

kamalcph replied:
Removed the method and fixed the other typographical errors.

kamalcph committed:
Utils#mkString empty params method removed

- mkString method is same as Map#toString. So, removed it.
- used place holder in logger
- typo error fixed

ijuma commented on clients/src/main/java/org/apache/kafka/common/utils/Utils.java:
```diff
@@ -372,7 +367,7 @@ public static String formatAddress(String host, Integer port) {
      * @param filename The path of the file to read
      */
-    public static Properties loadProps(String filename) throws IOException, FileNotFoundException {
+    public static Properties loadProps(String filename) throws IOException {
```
I'd keep `FileNotFoundException` in the signature as it indicates a specific error as opposed to the more general IOException

kamalcph replied:
Oh ok. I've remove it as intellj showed warning sign. I revert that change now.

kamalcph committed:
Unused import removed

kamalcph committed:
FileNotFoundException thrown together with IOException to be more specific while logging error

ijuma commented on clients/src/main/java/org/apache/kafka/common/utils/Utils.java:
```diff
@@ -524,17 +520,7 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
      * @param separator the separator
      */
     public static <T> CharSequence mkString(Collection<T> coll, String separator) {
```
I think we should remove this method since it's no different than `join` and is only used in tests.

kamalcph replied:
`mkString(coll, separator)` method removed.

kamalcph committed:
Utils#mkString(collection, separator) method removed

- It's usage are updated to Utils#join method

ijuma reviewed (approved):
Thanks, LGTM.
